### PR TITLE
Only update submodules that are in an uninitiated state

### PR DESCRIPTION
**Pull request by nhibberd**
In the current state this will `update` the submodule regardless of state.
This change will make sure it is only updated when the submodule is uninitiated state
@markhibberd @michaelneale @charleso

### CODE REVIEW

**Comment by michaelneale** [comment]
:8ball: I like it, however it is a change in behaviour to the current and previous behaviour? (which sounds like it wasn't really intended).

**Comment by nhibberd** [comment]
Yes it is different then both previous and current

File: src/cabal [hunk 1]
```diff
@@ -64,19 +64,12 @@ initialize () {
...
-        if [ -d "$submodule" ]; then
-           find "$submodule" -maxdepth 2 -name \*.cabal
-        fi
-    done | grep -q . || (
-        cd $PROJECT_ROOT
-        git submodule update --init
-    )
+        # only update and init submodules that are in uninitiated submodules
+        # this will work for sub-projects
+        git submodule update --init $submodule
```
Comment (michaelneale) [inline src/cabal]: I assume this works when you are deeper in the tree as well? as some things like sanction have cabal files deeper than their submodule deps.

**Comment by michaelneale** [comment]
@nhibberd oh no did I chase away the commit?

**Pull request closed without merging**